### ISSUE TITLE
Full support OsRTP (rfc8643)

### ISSUE THREAD

**Issue by mibofra** [signalwire/freeswitch#1254]
Hi! My bug is simple, and it is related to something here: https://lists.freeswitch.org/pipermail/freeswitch-users/2020-July/133675.html and here: https://lists.freeswitch.org/pipermail/freeswitch-users/2008-April/030760.html and here:
https://github.com/sipwise/rtpengine/issues/951 . With a semi-implementation with NDLB-allow-crypto-in-avp .

Can rfc8643 (OsRTP) be supported, with an option to fully enable it, when an extention or a gateway (or everything else) is using OsRTP and just offer OsRTP too? The thing is that, if this is not done, even if with NDLB-allow-crypto-in-avp, the other peer (an extention using optional srtp with rfc8643) answer with a 488 not acceptable here.

This is annoying, as for example, this make my yealink t23g (and I guess all the other yealink extentions too, with recent firmwares at least), unusable with optional srtp.

**Comment by mjerris** [comment]
optional crypto should be implemented as multiple m audio lines with and without crypto, not as crypto in avp

**Comment by mibofra** [comment]
I agree, but, as pointed out, there is an rfc, and unfortunally or not, it's used: https://datatracker.ietf.org/doc/html/rfc8643
So, do you (project) want to take the route of Google engineers forcing to don't use DHCPv6 on Android, with many users asking for it: https://issuetracker.google.com/issues/36949085 , or there is the possibility to implement it, specifically as an option, to support it?

On the other hand, would you accept pull requests to add support to rfc8643 in mainline freeswitch :D ?

**Comment by billsimon** [comment]
I have not seen any implemention of optional crypto as FreeSWITCH does it per mjerris. Agree with the philosophy, but Cisco et. al. won this one by implementing it differently. I vote also for rfc8643 in FS. @mibofra in the meantime you can front your freeswitch with opensips+rtpengine to get the kind of functionality you need.
